lib/exhal/document.ex

```diff
@@ -8,32 +8,56 @@ defmodule ExHal.Document do
 
   defstruct properties: %{}, links: %{}, client:
 
+
   @doc """
     Returns a new `%ExHal.Document` representing the HAL document provided.
     """
-  def parse(client, hal_str) do
+  def parse(hal_str, client \\ ExHal.client)
```

Comment:
it is called a "function head" and it's causes a compiler warning when there are multiple function clauses and only some of them have default values.